src/rbac/rbac.js

```diff
@@ -106,17 +108,21 @@ class RBAC {
     });
   }
 
-  addUser(userName, roleName) {
+  addUser(userName, roleNames) {
     if (this.getUser(userName)) throw new Error(`user already exists with username: ${userName}`);
 
-    if (!this.getRole(roleName)) throw new Error(`role doesnt exist: ${roleName}`);
+    if (!(roleNames instanceof Array)) throw new Error(`roles must be an array`);
```

Comment:
if we use the `roleNames` schema here (see comment about exporting it, we could easy check array type and accept single values to auto transfer into arrays (ie `single`) just using `joi`

Reply:
do we need a way to enforce not having duplicates in this array?